src/logcollector/read_multiline.c

```diff
@@ -93,7 +93,13 @@ void *read_multiline(logreader *lf, int *rc, int drop_it) {
             buffer_size++;
         }
 
-        strncpy(buffer + buffer_size, str, OS_MAXSTR - buffer_size - 2);
+        strncat(buffer, str, OS_MAXSTR - OS_LOG_HEADER - 1 - buffer_size);
```

Comment:
Note the ` - 1 ` I added. That is necessary because `strncat` will copy at most `n` bytes _from the source argument_ (where `n` is the third argument). 
Which means `strcat` could write a zero at buffer[sizeof buffer], which would be out of bounds.

Reply:
Hi @RamiroRD 

In the code it appears -1 for the same reason you say. 

strncat(buffer, str, **OS_MAXSTR - OS_LOG_HEADER - 1** - buffer_size);

Note that the size of the provided buffer is the size of the target buffer minus one because `strncat` copies up to n bytes and puts the character `n+1`. [Link](https://man7.org/linux/man-pages/man3/strcat.3.html)

> If src contains n or more bytes, strncat() writes n+1 bytes to dest (n from src plus the terminating null byte).  Therefore, the size of dest must be at least strlen(dest)+n+1.